### PR TITLE
[Frontend] Support for IsingZZ in Catalyst

### PR DESCRIPTION
**Pull request by mehrdad2m**
**Context:** We would like to add support for the PennyLane IsingZZ operation to our compilation architecture. The PennyLane-Lightning simulator supports this gate, and we would like to be able to support it without relying on the operation decomposition.

**Description of the Change:** Added IsingZZ to the list of supported gates. Added tests to confirm the functionality of IsingZZ gate and that the decomposition is not happening anymore.

**Benefits:** No need to decompose IsingZZ gate into a CNOT and RZ gate

**Possible Drawbacks:** May need more complicated tests to capture the full functionality of IsingZZ gate

**Related GitHub Issues:** #662

### CODE REVIEW

**Comment by erick-xanadu** [comment]
Hi @mehrdad2m ! Thanks for the submission. At the moment we have some failures on main that should be addressed by #729 . Please don't worry about errors related to `SProd` and `Sum` observables you see in your PR. :)

Cheers!

File: doc/changelog.md [hunk 1]
```diff
@@ -50,6 +50,11 @@
 
 <h3>Improvements</h3>
 
+* Added support for IsingZZ gate in Catalyst frontend. 
+Previously, the IsingZZ gate would be decomposed into 
+a CNOT and RZ gates. However, this is not needed as 
+the PennyLane-Lightning simulator supports this gate.
+
```
Comment (erick-xanadu) [inline doc/changelog.md:57]: Don't forget to add your name below :) I think this PR is good and we will likely merge it :)

Some small suggestions for the commit changelog message.

```suggestion
* Added support for IsingZZ gate in Catalyst frontend. 
  Previously, the IsingZZ gate would be decomposed into 
  a CNOT and RZ gates. However, this is not needed as 
  the PennyLane-Lightning simulator supports this gate.
  [(#730)](https://github.com/PennyLaneAI/catalyst/pull/730)
```

**Review by erick-xanadu — APPROVED** [review]
Hi @mehrdad2m !

This PR looks great! Regarding style, I think if you run `black -l 100 frontend/test/lit/test_multi_qubit_gates.py` that should fix it. We also have `make format` that would run `black`, `isort` and `pylint` along with any potential formatting in the C++ source code changes.

My only comment is regarding the changelog to add your name and add a url back to this PR.

Thanks!

**Comment by mehrdad2m** [comment]
> Hi @mehrdad2m !
> 
> This PR looks great! Regarding style, I think if you run `black -l 100 frontend/test/lit/test_multi_qubit_gates.py` that should fix it. We also have `make format` that would run `black`, `isort` and `pylint` along with any potential formatting in the C++ source code changes.
> 
> My only comment is regarding the changelog to add your name and add a url back to this PR.
> 
> Thanks!

Hi @erick-xanadu, Thanks for the review! I enjoyed working on it and getting familiar with the code base. Applied your suggestions and rebased the code, so hopefully, there would be no issues.

File: frontend/test/lit/test_multi_qubit_gates.py [hunk 2]
```diff
@@ -76,3 +76,16 @@ def circuit(x: float):
 
 
 print(circuit.mlir)
+
+
+# CHECK-LABEL: public @jit_isingZZ_circuit
+@qjit(target="mlir")
+@qml.qnode(qml.device("lightning.qubit", wires=2, shots=100))
+def isingZZ_circuit(x: float):
+    """Circuit that applies an IsingZZ gate to a pair of qbits."""
```
Comment (co9olguy) [inline frontend/test/lit/test_multi_qubit_gates.py:85]:
```suggestion
    """Circuit that applies an IsingZZ gate to a pair of qubits."""
```

Comment (mehrdad2m) [reply frontend/test/lit/test_multi_qubit_gates.py]: Done!

**Comment by erick-xanadu** [comment]
@mehrdad2m ignore the errors here, there are some errors introduced by a package upstream. Will merge when upstream is fixed.

**Comment by erick-xanadu** [comment]
@mehrdad2m I've updated the PR. It looks like code coverage for external contributor is not working well. I will chat about overriding that check to get this merge.

**Pull request merged**